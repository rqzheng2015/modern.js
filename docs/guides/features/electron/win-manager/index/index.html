<!doctype html>
<html class="docs-version-current" lang="zh-cn" dir="ltr">
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width,initial-scale=0.86,maximum-scale=3,minimum-scale=0.86">
<meta name="generator" content="Docusaurus v2.0.0-beta.6">
<link rel="alternate" type="application/rss+xml" href="/blog/rss.xml" title="Modern.js - 现代 Web 工程体系 Blog RSS Feed">
<link rel="alternate" type="application/atom+xml" href="/blog/atom.xml" title="Modern.js - 现代 Web 工程体系 Blog Atom Feed">

              <script>!function(e,r,n,t,s,o,a,i,c,l,d,f,m,p){o="precollect",a="getAttribute",i="addEventListener",(l=function(e){(d=[].slice.call(arguments)).push(Date.now(),location.href),(e==o?l.p.a:l.q).push(d)}).q=[],l.p={a:[]},e[s]=l,(f=document.createElement("script")).src=n+"?bid=modernjs_dev&globalName="+s,f.crossorigin=n.indexOf("sdk-web")>0?"anonymous":"use-credentials",r.getElementsByTagName("head")[0].appendChild(f),i in e&&(l.pcErr=function(r){r=r||e.event,(m=r.target||r.srcElement)instanceof Element||m instanceof HTMLElement?m[a]("integrity")?e[s](o,"sri",m[a]("href")||m[a]("src")):e[s](o,"st",{tagName:m.tagName,url:m[a]("href")||m[a]("src")}):e[s](o,"err",r.error)},l.pcRej=function(r){r=r||e.event,e[s](o,"err",r.reason||r.detail&&r.detail.reason)},e[i]("error",l.pcErr,!0),e[i]("unhandledrejection",l.pcRej,!0)),"PerformanceLongTaskTiming"in e&&((p=l.pp={entries:[]}).observer=new PerformanceObserver((function(e){p.entries=p.entries.concat(e.getEntries)})),p.observer.observe({entryTypes:["longtask","largest-contentful-paint","layout-shift"]}))}(window,document,"https://lf3-short.bytegoofy.com/slardar/fe/sdk-web/browser.cn.js",0,"Slardar")</script>
              

              <script>window.Slardar("init",{bid:"modernjs_dev"}),window.Slardar("start")</script><title data-react-helmet="true">介绍 | Modern.js - 现代 Web 工程体系</title><meta data-react-helmet="true" name="twitter:card" content="summary_large_image"><meta data-react-helmet="true" property="og:url" content="https://modernjs.dev//docs/guides/features/electron/win-manager/index"><meta data-react-helmet="true" name="docusaurus_locale" content="zh-cn"><meta data-react-helmet="true" name="docusaurus_version" content="current"><meta data-react-helmet="true" name="docusaurus_tag" content="docs-default-current"><meta data-react-helmet="true" property="og:title" content="介绍 | Modern.js - 现代 Web 工程体系"><meta data-react-helmet="true" name="description" content="窗口管理服务，主要帮助我们解决："><meta data-react-helmet="true" property="og:description" content="窗口管理服务，主要帮助我们解决："><link data-react-helmet="true" rel="shortcut icon" href="/img/favicon.ico"><link data-react-helmet="true" rel="canonical" href="https://modernjs.dev//docs/guides/features/electron/win-manager/index"><link data-react-helmet="true" rel="alternate" href="https://modernjs.dev//en/docs/guides/features/electron/win-manager/index" hreflang="en"><link data-react-helmet="true" rel="alternate" href="https://modernjs.dev//docs/guides/features/electron/win-manager/index" hreflang="zh-cn"><link data-react-helmet="true" rel="alternate" href="https://modernjs.dev//docs/guides/features/electron/win-manager/index" hreflang="x-default"><link rel="stylesheet" href="//lf-cdn-tos.bytescm.com/obj/static/webinfra/modern-js-website/assets/css/styles.73bb4626.css">
<link rel="preload" href="//lf-cdn-tos.bytescm.com/obj/static/webinfra/modern-js-website/assets/js/runtime~main.d05f735a.js" as="script">
<link rel="preload" href="//lf-cdn-tos.bytescm.com/obj/static/webinfra/modern-js-website/assets/js/main.3d6583ba.js" as="script">
</head>
<body itemscope="" itemtype="http://schema.org/Organization">
<script>!function(){function t(t){document.documentElement.setAttribute("data-theme",t)}var e=function(){var t=null;try{t=localStorage.getItem("theme")}catch(t){}return t}();t(null!==e?e:"light")}()</script><div id="__docusaurus">
<div><a href="#" class="skipToContent_V1Ig">跳转到主文档</a></div><nav class="navbar navbar--fixed-top navbarHideable_2J4B"><div class="navbar__inner"><div class="navbar__items"><button aria-label="Navigation bar toggle" class="navbar__toggle clean-btn" type="button" tabindex="0"><svg width="30" height="30" viewBox="0 0 30 30" aria-hidden="true"><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></button><a class="navbar__brand" href="/"><img src="https://lf-cdn-tos.bytescm.com/obj/static/webinfra/modern-js-website/assets/images/images/modernjs-logo.svg" alt="Modernjs Logo" class="themedImage_42gu themedImage--light_-FJF navbar__logo"><img src="https://lf-cdn-tos.bytescm.com/obj/static/webinfra/modern-js-website/assets/images/images/modernjs-logo.svg" alt="Modernjs Logo" class="themedImage_42gu themedImage--dark_p0fW navbar__logo"></a><a class="navbar__item navbar__link" href="/docs/start/mobile">开始</a><a class="navbar__item navbar__link navbar__link--active" href="/docs/guides/overview">指南</a><a class="navbar__item navbar__link" href="/docs/apis/overview">API</a><a class="navbar__item navbar__link" href="/blog">博客</a><a class="navbar__item navbar__link" href="/coming-soon">社区</a></div><div class="navbar__items navbar__items--right"><a href="https://github.com/modern-js-dev/modern.js" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link header-github-link" aria-label="GitHub repository"></a><div class="react-toggle toggle_91lq react-toggle--disabled"><div class="react-toggle-track" role="button" tabindex="-1"><div class="react-toggle-track-check"><span class="toggle_P1Jd">🌜</span></div><div class="react-toggle-track-x"><span class="toggle_P1Jd">🌞</span></div><div class="react-toggle-thumb"></div></div><input type="checkbox" class="react-toggle-screenreader-only" aria-label="Switch between dark and light mode"></div><div class="navbar__search searchBarContainer_0B71"><input placeholder="搜索" aria-label="Search" class="navbar__search-input"><div class="loadingRing_2Ffh searchBarLoadingRing_nyLg"><div></div><div></div><div></div><div></div></div></div></div></div><div role="presentation" class="navbar-sidebar__backdrop"></div></nav><div class="main-wrapper docs-wrapper docs-doc-page"><div class="docPage_aEDd"><button class="clean-btn backToTopButton_N4sD" type="button"><svg viewBox="0 0 24 24" width="28"><path d="M7.41 15.41L12 10.83l4.59 4.58L18 14l-6-6-6 6z" fill="currentColor"></path></svg></button><aside class="docSidebarContainer_mm6A"><div class="sidebar_khEP sidebarWithHideableNavbar_j1mR"><a tabindex="-1" class="sidebarLogo_2ggo" href="/"><img src="https://lf-cdn-tos.bytescm.com/obj/static/webinfra/modern-js-website/assets/images/images/modernjs-logo.svg" alt="Modernjs Logo" class="themedImage_42gu themedImage--light_-FJF"><img src="https://lf-cdn-tos.bytescm.com/obj/static/webinfra/modern-js-website/assets/images/images/modernjs-logo.svg" alt="Modernjs Logo" class="themedImage_42gu themedImage--dark_p0fW"></a><nav class="menu thin-scrollbar menu_2Snv menuWithAnnouncementBar_RH5t"><ul class="theme-doc-sidebar-menu menu__list"><li class="theme-doc-sidebar-item-link menu__list-item"><a class="menu__link" href="/docs/guides/overview">概览</a></li><li class="theme-doc-sidebar-item-category menu__list-item"><a class="menu__link menu__link--sublist" href="#">实战教程</a><ul style="display:block;overflow:visible;height:auto" class="menu__list"><li class="theme-doc-sidebar-item-link menu__list-item"><a class="menu__link" tabindex="0" href="/docs/guides/tutorials/overview">概览</a></li><li class="theme-doc-sidebar-item-category menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">1: 手动创建 MWA 项目</a></li><li class="theme-doc-sidebar-item-category menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">2: 自动创建 MWA 项目</a></li><li class="theme-doc-sidebar-item-category menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">3: 确认编程环境</a></li><li class="theme-doc-sidebar-item-category menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">4: 写 JS 代码</a></li><li class="theme-doc-sidebar-item-category menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">5: 使用 UI 组件</a></li><li class="theme-doc-sidebar-item-category menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">6: 添加 UI 组件</a></li><li class="theme-doc-sidebar-item-category menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">7: 添加应用入口</a></li><li class="theme-doc-sidebar-item-category menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">8: 添加客户端路由</a></li><li class="theme-doc-sidebar-item-category menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">9: 添加 BFF</a></li><li class="theme-doc-sidebar-item-category menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">10: 添加业务模型</a></li><li class="theme-doc-sidebar-item-category menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">11: 添加容器组件</a></li></ul></li><li class="theme-doc-sidebar-item-category menu__list-item"><a class="menu__link menu__link--sublist" href="#">更多用法</a><ul style="display:block;overflow:visible;height:auto" class="menu__list"><li class="theme-doc-sidebar-item-link menu__list-item"><a class="menu__link" tabindex="0" href="/docs/guides/usages/overview">概览</a></li><li class="theme-doc-sidebar-item-link menu__list-item"><a class="menu__link" tabindex="0" href="/docs/guides/usages/eslint">ESLint 全量规则集</a></li><li class="theme-doc-sidebar-item-category menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">CSS 开发方案</a></li><li class="theme-doc-sidebar-item-category menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">调试模式</a></li><li class="theme-doc-sidebar-item-category menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">常见配置</a></li><li class="theme-doc-sidebar-item-category menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">高级配置</a></li></ul></li><li class="theme-doc-sidebar-item-category menu__list-item"><a class="menu__link menu__link--sublist menu__link--active" href="#">专题详解</a><ul style="display:block;overflow:visible;height:auto" class="menu__list"><li class="theme-doc-sidebar-item-link menu__list-item"><a class="menu__link" tabindex="0" href="/docs/guides/features/overview">概览</a></li><li class="theme-doc-sidebar-item-category menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Runtime API</a></li><li class="theme-doc-sidebar-item-category menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">服务器端需求</a></li><li class="theme-doc-sidebar-item-category menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">微前端</a></li><li class="theme-doc-sidebar-item-category menu__list-item"><a class="menu__link menu__link--sublist menu__link--active" href="#" tabindex="0">Electron</a><ul style="display:block;overflow:visible;height:auto" class="menu__list"><li class="theme-doc-sidebar-item-link menu__list-item"><a class="menu__link" tabindex="0" href="/docs/guides/features/electron/basic">Electron 简介</a></li><li class="theme-doc-sidebar-item-link menu__list-item"><a class="menu__link" tabindex="0" href="/docs/guides/features/electron/modern-electron">Modern Electron 简介</a></li><li class="theme-doc-sidebar-item-link menu__list-item"><a class="menu__link" tabindex="0" href="/docs/guides/features/electron/develop">关于 Node 集成</a></li><li class="theme-doc-sidebar-item-category menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">通信</a></li><li class="theme-doc-sidebar-item-category menu__list-item"><a class="menu__link menu__link--sublist menu__link--active" href="#" tabindex="0">窗口管理</a><ul style="display:block;overflow:visible;height:auto" class="menu__list"><li class="theme-doc-sidebar-item-link menu__list-item"><a class="menu__link menu__link--active" aria-current="page" tabindex="0" href="/docs/guides/features/electron/win-manager/index">介绍</a></li><li class="theme-doc-sidebar-item-link menu__list-item"><a class="menu__link" tabindex="0" href="/docs/guides/features/electron/win-manager/open-window">打开窗口</a></li><li class="theme-doc-sidebar-item-link menu__list-item"><a class="menu__link" tabindex="0" href="/docs/guides/features/electron/win-manager/close-window">关闭窗口</a></li></ul></li><li class="theme-doc-sidebar-item-link menu__list-item"><a class="menu__link" tabindex="0" href="/docs/guides/features/electron/pack">构建</a></li><li class="theme-doc-sidebar-item-link menu__list-item"><a class="menu__link" tabindex="0" href="/docs/guides/features/electron/upgrade">升级</a></li><li class="theme-doc-sidebar-item-link menu__list-item"><a class="menu__link" tabindex="0" href="/docs/guides/features/electron/test">测试</a></li><li class="theme-doc-sidebar-item-category menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">其他</a></li></ul></li><li class="theme-doc-sidebar-item-category menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">可复用模块</a></li><li class="theme-doc-sidebar-item-category menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Monorepo</a></li><li class="theme-doc-sidebar-item-category menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">定制工程方案</a></li></ul></li></ul></nav></div></aside><main class="docMainContainer_Ip-H"><div class="container padding-top--md padding-bottom--lg"><div class="row"><div class="col docItemCol_JYLG"><div class="docItemContainer_Cd72"><article><div class="theme-doc-markdown markdown"><header><h1>介绍</h1></header><p>窗口管理服务，主要帮助我们解决：</p><ul><li>窗口的打开与关闭问题。</li><li>窗口之间或窗口与主进程的通信问题。</li><li>窗口的生命周期管理问题。</li></ul><p>具体内容如下：</p><ul><li><a href="/docs/guides/features/electron/win-manager/open-window">打开窗口</a></li><li><a href="/docs/guides/features/electron/win-manager/close-window">关闭窗口</a></li><li><a href="/docs/guides/features/electron/ipc/regist-services/render">窗口间通信</a></li><li>窗口管理 API<ul><li><a href="/docs/apis/runtime/electron/main-process/win-service">主进程 winService</a></li><li><a href="/docs/apis/runtime/electron/render-process/win-service">渲染进程 winService</a></li></ul></li></ul></div></article><nav class="pagination-nav docusaurus-mt-lg" aria-label="Docs pages navigation"><div class="pagination-nav__item"><a class="pagination-nav__link" href="/docs/guides/features/electron/ipc/regist-services/webview"><div class="pagination-nav__sublabel">上一篇</div><div class="pagination-nav__label">« webview 注册服务</div></a></div><div class="pagination-nav__item pagination-nav__item--next"><a class="pagination-nav__link" href="/docs/guides/features/electron/win-manager/open-window"><div class="pagination-nav__sublabel">下一篇</div><div class="pagination-nav__label">打开窗口 »</div></a></div></nav></div></div></div></div></main></div></div><footer class="footer footer--dark"><div class="container"><div class="row footer__links"><div class="col footer__col"><div class="footer__title">使用指南</div><ul class="footer__items"><ul class="footer__items"><li class="footer__item"><a class="footer__link-item" href="/docs/start/mobile">快速上手</a></li><li class="footer__item"><a class="footer__link-item" href="/docs/guides/tutorials/overview">实战教程</a></li><li class="footer__item"><a class="footer__link-item" href="/docs/guides/usages/eslint">更多用法</a></li><li class="footer__item"><a class="footer__link-item" href="/docs/guides/features/runtime/code-spliting">专题详解</a></li></ul></ul></div><div class="col footer__col"><div class="footer__title">API 资料</div><ul class="footer__items"><ul class="footer__items"><li class="footer__item"><a class="footer__link-item" href="/docs/apis/commands/overview">命令</a></li><li class="footer__item"><a class="footer__link-item" href="/docs/apis/runtime/overview">运行时</a></li><li class="footer__item"><a class="footer__link-item" href="/docs/apis/hooks/overview">文件约定</a></li><li class="footer__item"><a class="footer__link-item" href="/docs/apis/config/overview">配置选项</a></li></ul></ul></div><div class="col footer__col"><div class="footer__title">关注我们</div><ul class="footer__items"><ul class="footer__items footerItemIcons_L3RU"><a class="footer__item footerItemIcon_9u-Q"><img width="32" height="32" src="https://lf3-static.bytednsdoc.com/obj/eden-cn/aphqeh7uhohpquloj/modern-js/icons/weixin.png"><span class="qrcodeWrap_pGOd" style="display:none"><img width="108" height="108" src="https://lf3-static.bytednsdoc.com/obj/eden-cn/aphqeh7uhohpquloj/modern-js/icons/weixin-qrcode.jpeg" alt="微信"></span></a><a href="https://space.bilibili.com/1195398938" class="footer__item footerItemIcon_9u-Q"><img width="32" height="32" src="https://lf3-static.bytednsdoc.com/obj/eden-cn/aphqeh7uhohpquloj/modern-js/icons/bilibili.png"></a><a class="footer__item footerItemIcon_9u-Q"><img width="32" height="32" src="https://lf3-static.bytednsdoc.com/obj/eden-cn/aphqeh7uhohpquloj/modern-js/icons/feishu.png"><span class="qrcodeWrap_pGOd" style="display:none"><img width="108" height="108" src="https://lf3-static.bytednsdoc.com/obj/eden-cn/aphqeh7uhohpquloj/modern-js/icons/feishu-qrcode.png" alt="飞书"></span></a><a href="https://github.com/modern-js-dev/modern.js" class="footer__item footerItemIcon_9u-Q"><img width="32" height="32" src="https://lf3-static.bytednsdoc.com/obj/eden-cn/aphqeh7uhohpquloj/modern-js/icons/github.png"></a></ul></ul></div></div><div class="footer__bottom text--center"><div class="margin-bottom--sm"><img src="https://lf-cdn-tos.bytescm.com/obj/static/webinfra/modern-js-website/assets/images/images/modernjs-logo.svg" alt="ByteDance Open Source Logo" class="themedImage_42gu themedImage--light_-FJF footer__logo"><img src="https://lf-cdn-tos.bytescm.com/obj/static/webinfra/modern-js-website/assets/images/images/modernjs-logo.svg" alt="ByteDance Open Source Logo" class="themedImage_42gu themedImage--dark_p0fW footer__logo"></div></div></div></footer></div>
<script src="//lf-cdn-tos.bytescm.com/obj/static/webinfra/modern-js-website/assets/js/runtime~main.d05f735a.js"></script>
<script src="//lf-cdn-tos.bytescm.com/obj/static/webinfra/modern-js-website/assets/js/main.3d6583ba.js"></script>
</body>
</html>