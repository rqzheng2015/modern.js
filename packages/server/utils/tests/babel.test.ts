import path from 'path';
import { defaults } from './helpers';
import { resolveBabelConfig } from '@/index';

describe('babel', () => {
  it('resolveBabelConfig', () => {
    const pwd = path.resolve(__dirname, './fixtures');
    const tsconfigPath = path.resolve(
      __dirname,
      './fixtures/api/tsconfig.json',
    );
    const config = resolveBabelConfig(pwd, defaults as any, {
      type: 'commonjs',
      syntax: 'es6+',
      tsconfigPath,
    });

    const root = path.resolve(__dirname, '../../../../');

    expect.addSnapshotSerializer({
      test: val =>
        typeof val === 'string' &&
        (val.includes('modern.js') ||
          val.includes('node_modules') ||
          val.includes(root)),
      print: val =>
        // eslint-disable-next-line no-nested-ternary
        typeof val === 'string'
          ? // eslint-disable-next-line no-nested-ternary
            val.includes('node_modules')
            ? `"${val.replace(/.+node_modules/, '').replace(/\\/g, '/')}"`
            : val.includes('modern.js')
            ? `"${val.replace(/.+modern\.js/, '').replace(/\\/g, '/')}"`
            : `"${val.replace(root, '').replace(/\\/g, '/')}"`
          : (val as string),
    });

    expect(config).toMatchSnapshot();
  });
});
