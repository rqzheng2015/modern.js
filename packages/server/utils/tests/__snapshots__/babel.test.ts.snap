// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel resolveBabelConfig 1`] = `
Object {
  "plugins": Array [
    Array [
      "/@babel/plugin-transform-typescript/lib/index.js",
      Object {
        "allowDeclareFields": true,
      },
    ],
    Array [
      "/babel-plugin-macros/dist/index.js",
      Object {
        "twin": Object {
          "preset": "styled-components",
        },
      },
    ],
    Array [
      "/babel-plugin-import/lib/index.js",
      Object {
        "libraryDirectory": "es",
        "libraryName": "antd",
        "style": true,
      },
      "import-antd",
    ],
    Array [
      "/babel-plugin-lodash/lib/index.js",
      Object {
        "id": Array [
          "lodash",
          "ramda",
        ],
      },
    ],
    Array [
      "/@babel/plugin-proposal-decorators/lib/index.js",
      Object {
        "legacy": true,
      },
    ],
    Array [
      "/@babel/plugin-proposal-class-properties/lib/index.js",
      Object {
        "loose": true,
      },
    ],
    Array [
      "/@babel/plugin-proposal-private-methods/lib/index.js",
      Object {
        "loose": true,
      },
    ],
    Array [
      "/@babel/plugin-proposal-private-property-in-object/lib/index.js",
      Object {
        "loose": true,
      },
    ],
    Array [
      "/@babel/plugin-proposal-object-rest-spread/lib/index.js",
      Object {
        "useBuiltIns": true,
      },
    ],
    Array [
      "/@babel/plugin-transform-runtime/lib/index.js",
      Object {
        "corejs": false,
        "helpers": false,
        "regenerator": true,
        "useESModules": false,
        "version": "7.16.0",
      },
    ],
    Array [
      "/babel-plugin-transform-react-remove-prop-types/lib/index.js",
      Object {
        "removeImport": true,
      },
    ],
    Array [
      "/@babel/plugin-proposal-function-bind/lib/index.js",
    ],
    Array [
      "/@babel/plugin-proposal-export-default-from/lib/index.js",
    ],
    Array [
      "/@babel/plugin-proposal-export-namespace-from/lib/index.js",
    ],
    Array [
      "/@babel/plugin-proposal-optional-chaining/lib/index.js",
    ],
    Array [
      "/@babel/plugin-proposal-numeric-separator/lib/index.js",
    ],
    Array [
      "/@babel/plugin-proposal-pipeline-operator/lib/index.js",
      Object {
        "proposal": "minimal",
      },
    ],
    Array [
      "/@babel/plugin-proposal-partial-application/lib/index.js",
    ],
    Array [
      "/@babel/plugin-proposal-nullish-coalescing-operator/lib/index.js",
    ],
    Array [
      "/babel-plugin-styled-components/lib/index.js",
      Object {},
      "styled-components",
    ],
    Array [
      "/babel-plugin-transform-inline-environment-variables/lib/index.js",
      Object {
        "include": Array [],
      },
    ],
    Array [
      "/babel-plugin-module-resolver/lib/index.js",
      Object {
        "alias": Object {
          "@/*": Array [
            "./src/*",
          ],
        },
        "extensions": Array [
          ".ts",
          ".tsx",
          ".js",
          ".jsx",
          ".es",
          ".es6",
          ".mjs",
        ],
        "resolvePath": [Function],
        "root": "/packages/server/utils/tests/fixtures/",
      },
    ],
    Array [
      "/babel-plugin-transform-typescript-metadata/lib/plugin.js",
    ],
  ],
  "presets": Array [
    Array [
      "/@babel/preset-env/lib/index.js",
      Object {
        "bugfixes": true,
        "modules": "commonjs",
        "shippedProposals": false,
        "targets": Array [
          "chrome > 61",
          "edge > 16",
          "firefox > 60",
          "safari > 11",
          "ios_saf > 11",
        ],
      },
    ],
    Array [
      "/@babel/preset-react/lib/index.js",
      Object {
        "runtime": "automatic",
      },
    ],
    Array [
      "/@babel/preset-typescript/lib/index.js",
      Object {
        "allowDeclareFields": true,
        "allowNamespaces": true,
      },
    ],
  ],
}
`;
